FirebaseAppCheck/Sources/AppAttestProvider/API/FIRAppAttestAPIService.m

```diff
@@ -173,31 +206,50 @@ - (nullable NSData *)randomChallengeFromResponseBody:(NSData *)response error:(N
     return rejectedPromise;
   }
 
-  return [FBLPromise
-      onQueue:[self backgroundQueue]
-           do:^id _Nullable {
-             NSError *encodingError;
-             NSData *payloadJSON = [NSJSONSerialization dataWithJSONObject:@{
-               kRequestFieldKeyID : keyID,
-               kRequestFieldAttestation : [self base64StringWithData:attestation],
-               kRequestFieldChallenge : [self base64StringWithData:challenge]
-             }
-                                                                   options:0
-                                                                     error:&encodingError];
-
-             if (payloadJSON != nil) {
-               return payloadJSON;
-             } else {
-               return [FIRAppCheckErrorUtil JSONSerializationError:encodingError];
-             }
-           }];
+  return [FBLPromise onQueue:[self backgroundQueue]
+                          do:^id {
+                            id JSONObject = @{
+                              kRequestFieldKeyID : keyID,
+                              kRequestFieldAttestation : [self base64StringWithData:attestation],
+                              kRequestFieldChallenge : [self base64StringWithData:challenge]
+                            };
+
+                            return [self HTTPBodyWithJSONObject:JSONObject];
+                          }];
+}
+
+#pragma mark - Helpers
+
+- (id)HTTPBodyWithJSONObject:(nonnull id)JSONObject {
+  NSError *encodingError;
+  NSData *payloadJSON = [NSJSONSerialization dataWithJSONObject:JSONObject
+                                                        options:0
+                                                          error:&encodingError];
+  return payloadJSON ?: [FIRAppCheckErrorUtil JSONSerializationError:encodingError];
 }
```

Comment:
Added this helper to hopefully consolidate the JSON serialization. WDYT? Returning `id` to allow for either `NSData` or `NSError` but could instead return a promise that has been fulfilled/rejected...